src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/DescendantIterator.java

```diff
@@ -0,0 +1,100 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath.iterators;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+import net.sf.saxon.om.AxisInfo;
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.tree.iter.AxisIterator;
+import net.sf.saxon.tree.iter.SingleNodeIterator;
+
+/**
+ * Recursive-free implementation of the descendant axis iterator.
+ */
+public class DescendantIterator implements AxisIterator {
+
+    /**
+     * Enum defines starting node for iterator.
+     */
+    public enum StartWith {
+        /** Start with current node. */
+        CURRENT_NODE,
+        /** Omit current node and start with child nodes. */
+        CHILDREN
```

Comment:
Please add trailing `,`
This will allow to extend this enum in the future without changing this line.

Reply:
done

Reply:
We supposed to have Check on this.
I remember that it was long discussion PR somewhere

Reply:
@romani It was never finalized and merged. See https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/814 .